micropython/build.sh

```diff
@@ -4,41 +4,72 @@ set -e
 
 UPYTH_VER="1.15"
 UPYTH="micropython-${UPYTH_VER}"
+UPYTH_CC_BASE=$(echo "${CC}" | rev | cut -c4- | rev )
 
-b_log "Building micropython"
 PREFIX_UPYTH="${TOPDIR}/phoenix-rtos-ports/micropython"
 PREFIX_UPYTH_BUILD="${PREFIX_BUILD}/micropython"
-PREFIX_UPYTH_SRC=${PREFIX_UPYTH_BUILD}/${UPYTH}
 PREFIX_UPYTH_PATCHSET="${PREFIX_UPYTH}/${UPYTH}-patchset"
-PREFIX_UPYTH_MARKERS="$PREFIX_UPYTH_BUILD/markers/"
+
+b_log "Building micropython"
+pushd .
+cd "$PREFIX_UPYTH"
+
 
 #
-# Download and unpack
+# Download
 #
-mkdir -p "$PREFIX_UPYTH_BUILD" "$PREFIX_UPYTH_MARKERS"
-[ -f "$PREFIX_UPYTH/${UPYTH}.tar.xz" ] || wget https://github.com/micropython/micropython/releases/download/v${UPYTH_VER}/${UPYTH}.tar.xz -P "$PREFIX_UPYTH"
-[ -d "${PREFIX_UPYTH_SRC}" ] || tar xf "$PREFIX_UPYTH/${UPYTH}.tar.xz" -C "$PREFIX_UPYTH_BUILD"
+[ -f "${UPYTH}.tar.xz" ] || wget https://github.com/micropython/micropython/releases/download/v${UPYTH_VER}/${UPYTH}.tar.xz
+[ -d "${UPYTH}" ] || ( echo "Unpacking ${UPYTH}.tar.xz..." && tar xf "${UPYTH}.tar.xz" )
 
 
 #
-# Apply patches
+# Patch
+#
+if ! patch -R -p0 -s -f --dry-run < "${PREFIX_UPYTH_PATCHSET}"/lib.patch >/dev/null 2>&1
+then
+	patch -s -p0 < "${PREFIX_UPYTH_PATCHSET}"/lib.patch && echo "micropython build: lib/ patched"
+	patch -s -p0 < "${PREFIX_UPYTH_PATCHSET}"/py.patch&& echo "micropython build: py/ patched"
+	patch -s -p0 < "${PREFIX_UPYTH_PATCHSET}"/mpy-cross.patch && echo "micropython build: mpy-cross/ patched"
+	patch -s -p0 < "${PREFIX_UPYTH_PATCHSET}"/os.patch && echo "micropython build: ports/unix/ patched"
+fi
+
+
 #
-# FIXME: prefix patch files with numbers to ensure correct order
-for patchfile in "$PREFIX_UPYTH_PATCHSET"/*.patch; do
-	if [ ! -f "$PREFIX_UPYTH_MARKERS/$(basename "$patchfile").applied" ]; then
-		echo "applying patch: $patchfile"
-		patch -d "$PREFIX_UPYTH_SRC" -p1 < "$patchfile"   # FIXME: should be -p1
-		touch "$PREFIX_UPYTH_MARKERS/$(basename "$patchfile").applied"
-	fi
-done
+# Architecture specific flags/values
+#
+export PHOENIX_MATH_ABSENT="expm1 log1p asinh acosh atanh erf tgamma lgamma copysign __sin __cos __tan __signbit"
+if [[ ${UPYTH_CC_BASE} == "arm-phoenix-" ]]
+then
+	UPYTH_STACKSZ="2048"
+	UPYTH_HEAPSZ="16384"
+	export LDFLAGS_EXTRA=" -Wl,-q "
```

Comment:
same as below, probably invalid for 6ull, also, `max-page-size` should be defined for smaller executable. Maybe try to use provided LDFLAGS with some sed/awk to chage it to GCC format?

Reply:
As building/linking process and parameters changed drastically this part should be revised again